File: android-app/app/src/main/java/com/leafguard/database/ScanRecord.java
package com.leafguard.database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "scan_history")
public class ScanRecord {

    @PrimaryKey(autoGenerate = true)
    private long id;

    @ColumnInfo(name = "disease_name")
    private String diseaseName;

    @ColumnInfo(name = "confidence")
    private float confidence;

    @ColumnInfo(name = "symptoms")
    private String symptoms;

    @ColumnInfo(name = "treatment")
    private String treatment;

    @ColumnInfo(name = "prevention")
    private String prevention;

    @ColumnInfo(name = "image_uri")
    private String imageUri;

    @ColumnInfo(name = "latitude")
    private double latitude;

    @ColumnInfo(name = "longitude")
    private double longitude;

    @ColumnInfo(name = "timestamp")
    private long timestamp;

    public ScanRecord() {
    }

    @Ignore
    public ScanRecord(String diseaseName, float confidence, String symptoms, String treatment,
                      String prevention, String imageUri, double latitude, double longitude,
                      long timestamp) {
        this.diseaseName = diseaseName;
        this.confidence = confidence;
        this.symptoms = symptoms;
        this.treatment = treatment;
        this.prevention = prevention;
        this.imageUri = imageUri;
        this.latitude = latitude;
        this.longitude = longitude;
        this.timestamp = timestamp;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getDiseaseName() {
        return diseaseName;
    }

    public void setDiseaseName(String diseaseName) {
        this.diseaseName = diseaseName;
    }

    public float getConfidence() {
        return confidence;
    }

    public void setConfidence(float confidence) {
        this.confidence = confidence;
    }

    public String getSymptoms() {
        return symptoms;
    }

    public void setSymptoms(String symptoms) {
        this.symptoms = symptoms;
    }

    public String getTreatment() {
        return treatment;
    }

    public void setTreatment(String treatment) {
        this.treatment = treatment;
    }

    public String getPrevention() {
        return prevention;
    }

    public void setPrevention(String prevention) {
        this.prevention = prevention;
    }

    public String getImageUri() {
        return imageUri;
    }

    public void setImageUri(String imageUri) {
        this.imageUri = imageUri;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}

File: android-app-kotlin/app/src/main/java/com/leafguard/database/ScanRecord.kt
package com.leafguard.database

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Kotlin twin of ScanRecord.java.
 *
 * Identical Room schema: table name "scan_history" and the exact same
 * column names/types, so both apps produce byte-compatible databases.
 */
@Entity(tableName = "scan_history")
data class ScanRecord(
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0,

    @ColumnInfo(name = "disease_name")
    var diseaseName: String? = null,

    @ColumnInfo(name = "confidence")
    var confidence: Float = 0f,

    @ColumnInfo(name = "symptoms")
    var symptoms: String? = null,

    @ColumnInfo(name = "treatment")
    var treatment: String? = null,

    @ColumnInfo(name = "prevention")
    var prevention: String? = null,

    @ColumnInfo(name = "image_uri")
    var imageUri: String? = null,

    @ColumnInfo(name = "latitude")
    var latitude: Double = 0.0,

    @ColumnInfo(name = "longitude")
    var longitude: Double = 0.0,

    @ColumnInfo(name = "timestamp")
    var timestamp: Long = 0
)
